common/persistence/cassandra/nexus_service_store.go

```diff
@@ -61,30 +97,195 @@ func (s *NexusServiceStore) Close() {
 	}
 }
 
+func (s *NexusServiceStore) CreateOrUpdateNexusIncomingService(
+	ctx context.Context,
+	request *p.InternalCreateOrUpdateNexusIncomingServiceRequest,
+) error {
+	batch := s.session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
+
+	if request.Service.Version == 0 {
+		batch.Query(templateCreateIncomingServiceQuery,
+			constServicesPartition,
+			rowTypeIncomingNexusService,
+			request.Service.ServiceID,
+			request.Service.Data.Data,
+			request.Service.Data.EncodingType.String(),
+			0,
+		)
+	} else {
+		batch.Query(templateUpdateIncomingServiceQuery,
+			request.Service.Data.Data,
+			request.Service.Data.EncodingType.String(),
+			request.Service.Version+1,
+			constServicesPartition,
+			rowTypeIncomingNexusService,
+			request.Service.ServiceID,
+			request.Service.Version,
+		)
+	}
+
+	batch.Query(templateUpdateTableVersion,
```

Comment:
You need the same check here for `LastKnownTableVersion == 0` and issue an insert statement.